core/new-gui/src/app/common/service/user/user-dictionary/dictionary.service.ts

```diff
@@ -0,0 +1,518 @@
+   * and the proxy dictionary only changes if those requests are successful.
+   * *Warning: proxy dictionaries are synced w/ backend each getAll() and should be correct 99.99% of the time.
+   * However they may fall out of sync due to unforseen networking errors.*
+   * @returns Proxy of a local copy, Throws NotReadyError if initial getAll() request hasn't finished
+   */
+  public getUserDictionary(): UserDictionary {
+    if (!this.ready.value) { throw new NotReadyError('incomplete initialization of user-dictionary service'); }
+    return this.proxyUserDictionary();
+  }
+
+  /**
+   * get a proxy dictionary: access a local copy of the backend dictionary as an object *synchronously*
+   * without callbacks or subscriptions.
+   * All instances of proxy dictionaries are synced (changing one affects the others).
+   * assignment and deletion of attributes cause requests to be sent to the backend
+   * and the proxy dictionary only changes if those requests are successful.
+   * *Warning: proxy dictionaries are synced w/ backend each getAll() and should be correct 99.99% of the time.
+   * However they may fall out of sync due to unforeseen networking errors.*
+   * @returns Proxy of a local copy
+   */
+  public forceGetUserDictionary(): UserDictionary {
+    // gets user dictionary even if local dictionary isn't initialized
+    return this.proxyUserDictionary();
+  }
+
+  /**
+   * get a proxy dictionary: access a local copy of the backend dictionary as an object *synchronously*
+   * without callbacks or subscriptions.
+   * All instances of proxy dictionaries are synced (changing one affects the others).
+   * assignment and deletion of attributes cause requests to be sent to the backend
+   * and the proxy dictionary only changes if those requests are successful.
+   * *Warning: proxy dictionaries are synced w/ backend each getAll() and should be correct 99.99% of the time.
+   * However they may fall out of sync due to unforseen networking errors.*
+   * @returns Proxy of a local copy once initial getAll() is finished
+   */
+  public getUserDictionaryAsync(): Observable<UserDictionary> {
+    return from(this.ready.promise.then(() => this.getUserDictionary()));
+  }
+
+  /**
+   * initializes proxy dictionaries by calling getAll() and then setting dictionaryService.ready state
+   */
+  private initLocalDict() {
+    let resolveReady: (read: boolean) => void;
+
+    this.ready = {promise: new Promise((resolver) => resolveReady = resolver), value: false};
+
+    // getAll automatically creates a dictionaryEvent that updates the localdict with the remote
+    this.getAll().subscribe({
+      next: () => {
+        this.ready.value = true;
+        resolveReady(true);
+      },
+      error: (error: Error) => {
+        // user not logged in - print error but continue
+        if (error.message === 'Invalid session') {
+          console.error(error);
+        } else {
+          throw error;
+        }
```

Comment:
it appears different errors are being handled differently. what are the possible errors here? maybe for simplicity, just error out or warn out all the errors instead?

Reply:
I think warn out all errors would be better